game/qsp.ksy

```diff
@@ -0,0 +1,353 @@
+meta:
+  id: qsp
+  title: Quest Soft Player (QSP) Game
+  application: Quest Soft Player
+  file-extension: qsp
+  endian: le
+  encoding: "utf-16le" # in fact "ucs-2le"
+  license: Unlicense
+doc: |
+  QSP is a player for text quests written in a QSP language.
+
+doc-ref:
+  - https://github.com/QSPFoundation/qsp/blob/master/help/gam_desc.txt
+
+seq:
+  - id: signature
+    contents: ['Q',0, 'S',0, 'P',0, 'G',0, 'A',0, 'M',0, 'E',0, 0x0D,0, 0x0A,0] #QSPGAME
+    if: is_new_format
+  - id: version
+    type: usual_str
+    if:  is_new_format
+  - id: password
+    type: rot5_str
+    if:  is_new_format
+  - id: locations_count_str
+    type:
+      switch-on: is_new_format
+      cases:
+        true: rot5_number_str
+        false: number_str
+  - id: locations
+    type: loc
+    repeat: expr
+    repeat-expr: locations_count
+instances:
+  is_new_format:
+    value: true
```

Comment:
A mild suggestion: if you have some control over how it's called, it's generally a bad idea to name something "new". Today it's "old format" and "new format", tomorrow it will be "oldest format", "new-but-older format" and "newest format", and then it goes on. It's better to start enumerating them?

Reply:
Thanks, fixed. Assigned the formats versions numbers.